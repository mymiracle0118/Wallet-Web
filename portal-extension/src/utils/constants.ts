const CryptoJS = require('crypto-js')

export const shortenedAddress = (address: string) =>
  [address?.substring(0, 6), address?.substring(address.length - 4, address.length)].join('...')

export const handleCopy = (event: React.MouseEventHandler<HTMLButtonElement>) => {
  // eslint-disable-next-line
  event.stopPropagation()
}

export enum SAFETY_MEASURE {
  HIGH = 'high',
  MEDIUM = 'medium',
  LOW = 'low',
}

export const getWalletAddressRegex = (network: string) => {
  switch (network) {
    case 'APT':
    case 'SUI':
    case 'aptos':
    case 'SUPRA':
      return /^(0x)?[0-9A-Fa-f]{64}$/
    case 'SOL':
      return /^[1-9A-HJ-NP-Za-km-z]{32,}$/
    default:
      return /^0x[a-fA-F0-9]{40}$/
  }
}

export const remove0xStartOfString = (str: string) => {
  if (str.startsWith('0x') || str.startsWith('0X')) {
    str = str.substring(2)
  }
  return str
}

export const encryptData = (data: string, password: string) => {
  const encryptedData = CryptoJS.AES.encrypt(JSON.stringify(data) as string, password).toString()
  return encryptedData
}

export const decryptData = (encryptedData: string, password: string) => {
  const bytes = CryptoJS.AES.decrypt(encryptedData, password)
  const decryptedData = JSON.parse(bytes.toString(CryptoJS.enc.Utf8))
  return decryptedData
}

export enum TRANSACTION {
  COUNT = 15,
  NETWORK_FEE = 100,
  SUPRA_TOKEN_PRICE = 0.05,
}

export const add0xStartOfString = (str: string) => {
  if (!/^0x/i.test(str)) {
    return `0x${str}`
  }
  return str
}
