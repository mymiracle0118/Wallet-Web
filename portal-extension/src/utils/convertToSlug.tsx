export const convertToSlug = (str: string) => {
  return str
    .toLowerCase()
    .replace(/ /g, '-') // Replace spaces with dashes
    .replace(/[^\w-]+/g, '') // Remove non-word characters
}
