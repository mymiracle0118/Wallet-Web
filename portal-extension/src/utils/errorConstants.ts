interface errorTypes {
  [key: string]: string
}

const errorCoders: errorTypes = {
  UNKNOWN_ERROR: 'There is no way for network to know what the underlying problem is',
  UNSUPPORTED_OPERATION: 'The attempted operation is not supported',
  NETWORK_ERROR: 'Problem connecting to a network',
  BAD_DATA: 'Provided set of data cannot be correctly interpreted',
  CANCELLED: 'The operation was cancelled by a programmatic call',
  BUFFER_OVERRUN: 'An attempt was made to read outside the bounds of protected data',
  NUMERIC_FAULT: 'An operation which would result in incorrect arithmetic output has occurred',
  CALL_EXCEPTION: 'Transaction reverted',
  REPLACEMENT_UNDERPRICED:
    'An attempt was made to replace a transaction, but with an insufficient additional fee to afford evicting the old transaction from the memory pool',
  TRANSACTION_REPLACED: 'A pending transaction was replaced by another',
  UNCONFIGURED_NAME: 'An ENS name was used, but the name has not been configured',
  OFFCHAIN_FAULT: 'A CCIP-read exception, which cannot be recovered from or be further processed',
  ACTION_REJECTED: 'This Error indicates a request was rejected by the user',
  INSUFFICIENT_FUNDS: 'Insufficient balance for transaction fee',
  ERR_BAD_RESPONSE: 'Request failed with status code 500',
  NONCE_EXPIRED: 'Transaction already completed',
  INVALID_ARGUMENT: 'Invalid address',
  ECONNABORTED: 'Timeout limit exceeded',
  MISSING_ARGUMENT: 'Missing argument',
  UNEXPECTED_ARGUMENT: 'Unexpected argument',
  INVALID_TRANSACTION: 'Invalid transaction',
  INSUFFICIENT_TX_FEE: 'Insufficient transaction fee',
  STATIC_CALL_FAILED: 'Static call failed',
  GAS_ESTIMATE: 'Gas estimate failed',
  UNPREDICTABLE_GAS_LIMIT: 'Gas limit estimation failed',
  UNPREDICTABLE_GAS_PRICE: 'Gas price estimation failed',
  INVALID_NETWORK: 'Invalid network',
  INVALID_RPC_RESPONSE: 'Invalid RPC response',
  TRANSACTION_FAILED: 'Transaction failed',
  TIMEOUT: 'Operation timed out',
  CLIENT_ERROR: 'Client error',
  SERVER_ERROR: 'Server error',
  RATE_LIMIT: 'Rate limit exceeded',
  PROVIDER_ERROR: 'Provider error',
  MISSING_RATE_LIMIT_HEADER: 'Missing rate limit header',
  INSUFFICIENT_POLLING: 'Insufficient polling',
  JSON_RPC_REQUEST_FAILED: 'JSON RPC request failed',
  CONNECTION_TIMEOUT: 'Connection timeout',
  POOR_RESPONSE_TIME: 'Poor response time',
  INSUFFICIENT_DATA: 'Insufficient data',
  QUERY_RESPONDED_WITH_ERROR: 'Query responded with error',
  NO_RELAY_SET: 'No relay set',

  UNKNOWN_VALIDATION_STATUS: 'Transaction status determination error',
  INVALID_SIGNATURE: 'Invalid transaction signature',
  INVALID_AUTH_KEY: 'Bad account authentication key',
  SEQUENCE_NUMBER_TOO_OLD: 'Outdated sequence number',
  SEQUENCE_NUMBER_TOO_NEW: 'New sequence number',
  INSUFFICIENT_BALANCE_FOR_TRANSACTION_FEE: 'Insufficient balance for transaction fee',
  TRANSACTION_EXPIRED: 'Expired transaction',
  SENDING_ACCOUNT_DOES_NOT_EXIST: 'Non-existent sending account',
  REJECTED_WRITE_SET: 'Write set transaction rejected',
  INVALID_WRITE_SET: 'Invalid write set transaction',
  EXCEEDED_MAX_TRANSACTION_SIZE: 'Transaction size exceeds limit',
  UNKNOWN_SCRIPT: 'Unknown script error',
  UNKNOWN_MODULE: 'Attempt to publish a new module',
  MAX_GAS_UNITS_EXCEEDS_MAX_GAS_UNITS_BOUND: 'Max gas units exceed VM limit',
  MAX_GAS_UNITS_BELOW_MIN_TRANSACTION_GAS_UNITS: 'Max gas units below transaction cost',
  GAS_UNIT_PRICE_BELOW_MIN_BOUND: 'Gas unit price below minimum set in VM',
  GAS_UNIT_PRICE_ABOVE_MAX_BOUND: 'Gas unit price above maximum set in VM',
  INVALID_GAS_SPECIFIER: 'Invalid gas specifier',
  SENDING_ACCOUNT_FROZEN: 'Frozen sending account',
  UNABLE_TO_DESERIALIZE_ACCOUNT: 'Account deserialization error',
  CURRENCY_INFO_DOES_NOT_EXIST: 'Currency info not found',
  INVALID_MODULE_PUBLISHER: 'Insufficient permissions to publish modules',
  NO_ACCOUNT_ROLE: 'Sending account has no role',
  BAD_CHAIN_ID: 'Incorrect chain ID',
  SEQUENCE_NUMBER_TOO_BIG: 'Sequence number too large',
  BAD_TRANSACTION_FEE_CURRENCY: 'Gas currency not registered',
  FEATURE_UNDER_GATING: 'Feature intended for future version',
  SECONDARY_KEYS_ADDRESSES_COUNT_MISMATCH: 'Mismatch in secondary signer addresses',
  SIGNERS_CONTAIN_DUPLICATES: 'Duplicate signers detected',
  SEQUENCE_NONCE_INVALID: 'Invalid sequence nonce',
  CHAIN_ACCOUNT_INFO_DOES_NOT_EXIST: 'Error accessing chain-specific account info',
  MODULE_ADDRESS_DOES_NOT_MATCH_SENDER: "Module address doesn't match sender",
  DUPLICATE_MODULE_NAME: 'Attempt to publish two modules with the same name',
  BACKWARD_INCOMPATIBLE_MODULE_UPDATE: 'Module update breaks compatibility',
  CYCLIC_MODULE_DEPENDENCY: 'Cyclic dependency in module update',
  RESOURCE_DOES_NOT_EXIST: 'Resource not found under the account',
  RESOURCE_ALREADY_EXISTS: 'Attempt to create an existing resource',
  UNKNOWN_STATUS: 'Unknown error',
}

export const getErrorMessage = (errorCode: string) => {
  return errorCoders[errorCode] || 'Something went wrong'
}
