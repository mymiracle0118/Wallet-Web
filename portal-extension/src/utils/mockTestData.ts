export const ethereumAddress = '0x63C100ac0C36549C7218070294b60C18d813675c'

export const mockAssets = [
  {
    id: 'ethereum',
    network: 'mainnet',
    chain: 'ethereum',
    type: 'layer1',
    name: 'Ethereum',
    symbol: 'ETH',
    decimal: 18,
    image: 'https://assets.coingecko.com/coins/images/279/large/ethereum.png',
    enabledAsDefault: true,
  },
  {
    id: 'ethereum',
    network: 'goerli',
    chain: 'ethereum',
    type: 'layer1',
    name: 'GoerliETH',
    symbol: 'ETH',
    decimal: 18,
    image: 'https://assets.coingecko.com/coins/images/279/large/ethereum.png',
    enabledAsDefault: false,
  },
]

export const mockAccounts = {
  '0x63C100ac0C36549C7218070294b60C18d813675c': {
    username: 'Account 1',
    address: '0x63C100ac0C36549C7218070294b60C18d813675c',
    assets: [
      {
        chain: 'ethereum',
        decimal: 18,
        enabledAsDefault: true,
        formattedBalance: 0.000307848568706,
        id: 'ethereum',
        image: 'https://assets.coingecko.com/coins/images/279/large/ethereum.png',
        name: 'Ethereum',
        network: 'mainnet',
        symbol: 'ETH',
        type: 'layer1',
      },
      {
        chain: 'ethereum',
        decimal: 18,
        enabledAsDefault: true,
        formattedBalance: 2.034722077599812,
        id: 'ethereum',
        image: 'https://assets.coingecko.com/coins/images/279/large/ethereum.png',
        name: 'GoerliETH',
        network: 'goerli',
        symbol: 'ETH',
        type: 'layer1',
      },
    ],
  },
  '0x1Aa2C1c6799d320c57a0613F1F66Ea9590257bA1': {
    address: '0x1Aa2C1c6799d320c57a0613F1F66Ea9590257bA1',
    username: 'Account2',
    assets: [
      {
        chain: 'ethereum',
        decimal: 18,
        enabledAsDefault: true,
        formattedBalance: 0,
        id: 'ethereum',
        image: 'https://assets.coingecko.com/coins/images/279/large/ethereum.png',
        name: 'Ethereum',
        network: 'mainnet',
        symbol: 'ETH',
        type: 'layer1',
      },
      {
        chain: 'ethereum',
        decimal: 18,
        enabledAsDefault: true,
        formattedBalance: 0,
        id: 'ethereum',
        image: 'https://assets.coingecko.com/coins/images/279/large/ethereum.png',
        name: 'GoerliETH',
        network: 'goerli',
        symbol: 'ETH',
        type: 'layer1',
      },
    ],
  },
}

export const mockAssetsWithSUI = [
  {
    id: 'ethereum',
    network: 'mainnet',
    chain: 'ethereum',
    type: 'layer1',
    name: 'Ethereum',
    symbol: 'ETH',
    decimal: 18,
    image: 'https://assets.coingecko.com/coins/images/279/large/ethereum.png',
    enabledAsDefault: true,
  },
  {
    id: 'ethereum',
    network: 'goerli',
    chain: 'ethereum',
    type: 'layer1',
    name: 'GoerliETH',
    symbol: 'ETH',
    decimal: 18,
    image: 'https://assets.coingecko.com/coins/images/279/large/ethereum.png',
    enabledAsDefault: false,
  },
  {
    id: 'SUI',
    network: 'SUI',
    chain: 'SUI',
    name: 'SUI',
    symbol: 'SUI',
    decimal: 0,
    image: 'https://raw.githubusercontent.com/MystenLabs/sui/main/doc/static/Sui_Icon_Brand.png',
    enabledAsDefault: false,
  },
]

export const mockNetworkAssets = [
  {
    blockExplorer: 'https://goerli.etherscan.io',
    chainId: 5,
    coin: 'Goerli',
    enabledAsDefault: true,
    image: 'https://assets.coingecko.com/coins/images/279/large/ethereum.png',
    name: 'Goerli Test Network',
    networkId: 'goerli',
    symbol: 'GoerliETH',
    testNetwork: true,
  },
]

export const mockTestProfit = [
  {
    network: 'goerli',
    time: 'Mon Sep 26 2022 20:49:21 GMT+0800 (Philippine Standard Time) {}',
    token: 'ETH',
    value: 2.524722077554242,
  },
  {
    network: 'mainnet',
    time: 'Mon Sep 26 2022 20:49:22 GMT+0800 (Philippine Standard Time) {}',
    token: 'ETH',
    value: 0.000307848568706,
  },
]
