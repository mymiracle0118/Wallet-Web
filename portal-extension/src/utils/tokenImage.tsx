import React from 'react'

import ETH from 'assets/coins/ETH.svg'
import USDC from 'assets/coins/USDC.svg'
import SUSHI from 'assets/coins/SUSHI.svg'
import UNI from 'assets/coins/SUP.svg'
import USDT from 'assets/coins/USDT.svg'
import WETH from 'assets/coins/WETH.svg'
import SUI from 'assets/coins/SUI.svg'
import APTOS from 'assets/coins/APTOS.svg'
import SOL from 'assets/coins/SOL.svg'
import SUP from 'assets/coins/SUP.svg'

export const tokenImage = (symbol: string) => {
  switch (symbol) {
    case 'ETH':
      return <ETH />
    case 'USDC':
      return <USDC />
    case 'SUSHI':
      return <SUSHI />
    case 'UNI':
      return <UNI />
    case 'WETH':
      return <WETH />
    case 'USDT':
      return <USDT />
    case 'SUI':
      return <SUI />
    case 'APTOS':
      return <APTOS />
    case 'SOL':
      return <SOL />
    case 'SUPRA':
      return <SUP />
    default:
      return <ETH />
  }
}
