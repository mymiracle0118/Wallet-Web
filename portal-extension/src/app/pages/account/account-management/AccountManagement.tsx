import React, { useEffect, useMemo, useState } from 'react'
import { useTranslation } from 'react-i18next'
import SinglePageTitleLayout from 'layouts/single-page-layout/SinglePageLayout'
import AccountItem from './AccountItem'
import { useNavigate } from 'lib/woozie'
import { useSettings } from '@portal/shared/hooks/useSettings'
import { useWallet } from '@portal/shared/hooks/useWallet'
import { ethers } from 'ethers'
import { usePricing } from '@portal/shared/hooks/usePricing'
import { Listbox, ListboxItem } from '@nextui-org/react'
import { IAccountItemProps } from '@portal/shared/utils/types'
import { Button } from 'components'
import { KeyIcon, PlusIcon } from '@src/app/components/Icons'
import { useStore } from '@portal/shared/hooks/useStore'

const AccountManagement = () => {
  const { t } = useTranslation()
  const { navigate } = useNavigate()
  const { accounts, currentAccount, setCurrentAccount } = useSettings()
  const { setLockWallet, isAccountCreatedByPrivateKey } = useWallet()

  // TODO :: saveAccount reseting default assets with balance check another option to show balance
  const handleChangeActiveAccount = (value: string) => {
    setCurrentAccount(accounts[value])
    navigate('/home')
  }
  const { getAssetValue } = usePricing()
  const { currentTokenArrayWithBalance } = useStore()
  const [totalBalance, setTotalBalance] = useState<number>(0)
  useEffect(() => {
    let total = 0
    Object.values(currentTokenArrayWithBalance).forEach((asset) => {
      const assetBalance = Number(asset.formattedBalance || 0)
      total += getAssetValue(asset.coingeckoTokenId, assetBalance)
    })

    setTotalBalance(total)
  }, [currentTokenArrayWithBalance, totalBalance, getAssetValue])

  const accountData = useMemo(
    () =>
      Object.values(accounts).map<IAccountItemProps>((data) => ({
        address: data.address,
        userName: data.username,
        avatar: data.avatar,
        balance: 0,
        isAccountImported: data.isAccountImported,
        active: data.address === currentAccount?.address,
      })),
    [accounts]
  )

  return (
    <SinglePageTitleLayout title={t('Account.accountTitle')}>
      <div className="flex flex-col h-full">
        <div className="flex-1 space-y-5">
          <div className={'p-2 bg-surface-dark-alt rounded-lg'}>
            {accountData.map((data, idx) => (
              <AccountItem
                key={idx}
                active={data.active}
                avatar={data.avatar}
                address={data.address}
                userName={data.userName}
                balance={data.active ? `$${ethers.utils.commify(totalBalance.toFixed(2))}` : ''}
                onClick={handleChangeActiveAccount}
                isAccountImported={data.isAccountImported}
              />
            ))}
          </div>

          <div className="border-small px-1 py-2 rounded-xl dark:border-custom-white10">
            <Listbox aria-label="Account" variant="flat">
              {!isAccountCreatedByPrivateKey ? (
                <ListboxItem
                  key="create_sub_Account"
                  startContent={<PlusIcon className="mr-2" />}
                  className="font-extrabold py-3"
                  onClick={() => navigate('/account/create-sub-account')}
                >
                  {t('Account.createSubAccount')}
                </ListboxItem>
              ) : null}
              <ListboxItem
                key="import_Account"
                startContent={<KeyIcon className="mr-2" />}
                className="font-extrabold py-3"
                onClick={() => navigate('/account/import-by-private-key')}
              >
                {t('Account.importAccount')}
              </ListboxItem>
            </Listbox>
          </div>
        </div>
        <div className="pt-4 flex">
          <Button
            type="button"
            data-aid="logOutButton"
            onClick={() => {
              setLockWallet(true), navigate('/login')
            }}
            variant="bordered"
            color="outlined"
          >
            {t('Actions.logout')}
          </Button>
        </div>
      </div>
    </SinglePageTitleLayout>
  )
}

export default AccountManagement
