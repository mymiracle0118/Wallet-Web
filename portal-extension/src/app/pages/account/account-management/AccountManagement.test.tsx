import React from 'react'
import { render, renderHook, act, screen } from '@testing-library/react'
import AccountManagement from 'pages/account/account-management/AccountManagement'
import { useWallet } from '@portal/shared/hooks/useWallet'
import { useSettings } from '@portal/shared/hooks/useSettings'
import userEvent from '@testing-library/user-event'
import { I18nextProvider } from 'react-i18next'
import i18n from '@portal/shared/i18n/config'
import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
import { mockAccounts } from 'utils/mockTestData'

describe('Account Management Page UI', () => {
  const queryClient = new QueryClient()

  test('renders correctly Account Management Page', () => {
    const tree = render(
      <QueryClientProvider client={queryClient}>
        <AccountManagement />
      </QueryClientProvider>
    )
    expect(tree).toMatchSnapshot()
  })

  beforeEach(() => {
    const { result: walletResult } = renderHook(() => useWallet())
    const { result: walletSettings } = renderHook(() => useSettings())
    act(() => {
      walletResult.current.address = '0x63C100ac0C36549C7218070294b60C18d813675c'
      walletResult.current.username = 'Account 1'

      walletSettings.current.accounts = mockAccounts
    })
  })

  test('Change account', async () => {
    const { result: walletResult } = renderHook(() => useWallet())
    const user = userEvent.setup()
    render(
      <QueryClientProvider client={queryClient}>
        <I18nextProvider i18n={i18n}>
          <AccountManagement />
        </I18nextProvider>
      </QueryClientProvider>
    )

    await user.click(screen.getByTestId('account-list-Account2'))
    expect(walletResult.current.address).toBe('0x1Aa2C1c6799d320c57a0613F1F66Ea9590257bA1')
    expect(walletResult.current.username).toBe('Account2')
  })

  test('Hide account', async () => {
    const { result: walletResult } = renderHook(() => useWallet())
    const { result: walletSettings } = renderHook(() => useSettings())
    const user = userEvent.setup()
    render(
      <QueryClientProvider client={queryClient}>
        <AccountManagement />
      </QueryClientProvider>
    )
    await user.click(screen.getByTestId('Account2-more-options'))
    await user.click(screen.getByTestId('open-hide-account-modal'))
    await user.click(screen.getByTestId('button-hide-account'))
    expect(walletResult.current.username).toBe('Account 1')
    expect(walletSettings.current.accounts['0x1Aa2C1c6799d320c57a0613F1F66Ea9590257bA1']).toBeUndefined()
  })
})
