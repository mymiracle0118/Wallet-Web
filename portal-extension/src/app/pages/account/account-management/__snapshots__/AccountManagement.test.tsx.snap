// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account Management Page UI renders correctly Account Management Page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
    <div />
    <div>
      <div
        class="dark:bg-surface-dark bg-custom-white flex flex-col h-[600px] dark:text-custom-white text-custom-black"
      >
        <div
          class="flex flex-row relative justify-between items-center h-16"
        >
          <button
            class="h-[24px] w-[24px] p-1 ml-1 hover:bg-custom-grey10 dark:hover:bg-custom-white10 rounded-full flex items-center justify-center absolute"
            type="button"
          >
            <arrow-left.svg
              classname="text-xl stroke-custom-black dark:stroke-custom-white"
            />
          </button>
          <div
            class="flex-1 text-center"
          >
            <div
              class="text-custom-black dark:text-custom-white text-lg font-extrabold"
            >
              Account.accountTitle
            </div>
          </div>
        </div>
        <div
          class="w-full h-full overflow-auto"
        >
          <div
            class="flex flex-col h-full"
          >
            <div
              class="flex-1"
            >
              <div
                class="active MuiBox-root css-tb33on"
              >
                <div
                  class="css-rpfcki"
                >
                  <svg
                    class="inline-block select-none shrink-0 fill-custom-black dark:fill-custom-white text-4xl"
                    style="height: 1em; width: 1em;"
                  >
                    <logo.svg />
                  </svg>
                  <div
                    class="badge-check"
                  >
                    <check.svg />
                  </div>
                </div>
                <button
                  class="flex flex-col w-full"
                  data-testid="account-list-Account 1"
                  type="button"
                >
                  <div
                    class="text-custom-black dark:text-custom-white text-sm font-extrabold"
                    style="flex: 1; margin-left: 8px;"
                  >
                    Account 1
                  </div>
                  <div
                    class="text-custom-black dark:text-custom-white text-sm font-regular"
                    style="flex: 1; margin-left: 8px;"
                  >
                    $
                    0
                  </div>
                </button>
                <button
                  aria-expanded="false"
                  aria-haspopup="true"
                  class="relative"
                  data-testid="Account 1-more-options"
                  id="headlessui-listbox-button-:r0:"
                  type="button"
                >
                  <button
                    class="rounded-full p-2 hover:bg-custom-grey10 dark:hover:bg-custom-white10 sort-btn"
                    type="button"
                  >
                    <more-horizontal.svg />
                  </button>
                </button>
                <div
                  style="position: absolute; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px; display: none;"
                />
              </div>
              <div
                class=" MuiBox-root css-tb33on"
              >
                <div
                  class="css-rpfcki"
                >
                  <svg
                    class="inline-block select-none shrink-0 fill-custom-black dark:fill-custom-white text-4xl"
                    style="height: 1em; width: 1em;"
                  >
                    <logo.svg />
                  </svg>
                </div>
                <button
                  class="flex flex-col w-full"
                  data-testid="account-list-Account2"
                  type="button"
                >
                  <div
                    class="text-custom-black dark:text-custom-white text-sm font-extrabold"
                    style="flex: 1; margin-left: 8px;"
                  >
                    Account2
                  </div>
                  <div
                    class="text-custom-black dark:text-custom-white text-sm font-regular"
                    style="flex: 1; margin-left: 8px;"
                  >
                    $
                    0
                  </div>
                </button>
                <button
                  aria-expanded="false"
                  aria-haspopup="true"
                  class="relative"
                  data-testid="Account2-more-options"
                  id="headlessui-listbox-button-:r2:"
                  type="button"
                >
                  <button
                    class="rounded-full p-2 hover:bg-custom-grey10 dark:hover:bg-custom-white10 sort-btn"
                    type="button"
                  >
                    <more-horizontal.svg />
                  </button>
                </button>
                <div
                  style="position: absolute; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px; display: none;"
                />
              </div>
              <div
                class="MuiBox-root css-ytqq2v"
              >
                <div
                  class="css-tac3yh"
                >
                  <svg
                    class="inline-block select-none shrink-0 fill-custom-black dark:fill-custom-white text-4xl"
                    style="height: 1em; width: 1em;"
                  >
                    <plus-circle.svg />
                  </svg>
                </div>
                <div
                  class="MuiBox-root css-hboir5"
                >
                  <div
                    class="text-custom-black dark:text-custom-white text-sm font-extrabold"
                    style="flex: 1; margin-left: 8px;"
                  >
                    Account.createSubAccount
                  </div>
                </div>
              </div>
              <div
                class="MuiBox-root css-ytqq2v"
              >
                <div
                  class="css-tac3yh"
                >
                  <svg
                    class="inline-block select-none shrink-0 fill-custom-black dark:fill-custom-white text-4xl"
                    style="height: 1em; width: 1em;"
                  >
                    <key.svg />
                  </svg>
                </div>
                <div
                  class="MuiBox-root css-hboir5"
                >
                  <div
                    class="text-custom-black dark:text-custom-white text-sm font-extrabold"
                    style="flex: 1; margin-left: 8px;"
                  >
                    Account.importAccount
                  </div>
                </div>
              </div>
            </div>
            <div
              class="p-4 flex"
            >
              <button
                class="w-full border-[1px] border-solid border-primary dark:border-custom-white40 dark:hover:border-primary rounded-xl profile-btn flex justify-center items-center mr-1 dark:text-custom-white font-bold hover:bg-custom-grey10 dark:hover:bg-transparent p-3 rounded-md text-custom-black dark:text-custom-white"
                data-aid="logOutButton"
                type="button"
              >
                Actions.logout
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      id="headlessui-portal-root"
    >
      <div
        data-headlessui-portal=""
      >
        <div />
      </div>
      <div
        data-headlessui-portal=""
      >
        <div />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="dark:bg-surface-dark bg-custom-white flex flex-col h-[600px] dark:text-custom-white text-custom-black"
    >
      <div
        class="flex flex-row relative justify-between items-center h-16"
      >
        <button
          class="h-[24px] w-[24px] p-1 ml-1 hover:bg-custom-grey10 dark:hover:bg-custom-white10 rounded-full flex items-center justify-center absolute"
          type="button"
        >
          <arrow-left.svg
            classname="text-xl stroke-custom-black dark:stroke-custom-white"
          />
        </button>
        <div
          class="flex-1 text-center"
        >
          <div
            class="text-custom-black dark:text-custom-white text-lg font-extrabold"
          >
            Account.accountTitle
          </div>
        </div>
      </div>
      <div
        class="w-full h-full overflow-auto"
      >
        <div
          class="flex flex-col h-full"
        >
          <div
            class="flex-1"
          >
            <div
              class="active MuiBox-root css-tb33on"
            >
              <div
                class="css-rpfcki"
              >
                <svg
                  class="inline-block select-none shrink-0 fill-custom-black dark:fill-custom-white text-4xl"
                  style="height: 1em; width: 1em;"
                >
                  <logo.svg />
                </svg>
                <div
                  class="badge-check"
                >
                  <check.svg />
                </div>
              </div>
              <button
                class="flex flex-col w-full"
                data-testid="account-list-Account 1"
                type="button"
              >
                <div
                  class="text-custom-black dark:text-custom-white text-sm font-extrabold"
                  style="flex: 1; margin-left: 8px;"
                >
                  Account 1
                </div>
                <div
                  class="text-custom-black dark:text-custom-white text-sm font-regular"
                  style="flex: 1; margin-left: 8px;"
                >
                  $
                  0
                </div>
              </button>
              <button
                aria-expanded="false"
                aria-haspopup="true"
                class="relative"
                data-testid="Account 1-more-options"
                id="headlessui-listbox-button-:r0:"
                type="button"
              >
                <button
                  class="rounded-full p-2 hover:bg-custom-grey10 dark:hover:bg-custom-white10 sort-btn"
                  type="button"
                >
                  <more-horizontal.svg />
                </button>
              </button>
              <div
                style="position: absolute; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px; display: none;"
              />
            </div>
            <div
              class=" MuiBox-root css-tb33on"
            >
              <div
                class="css-rpfcki"
              >
                <svg
                  class="inline-block select-none shrink-0 fill-custom-black dark:fill-custom-white text-4xl"
                  style="height: 1em; width: 1em;"
                >
                  <logo.svg />
                </svg>
              </div>
              <button
                class="flex flex-col w-full"
                data-testid="account-list-Account2"
                type="button"
              >
                <div
                  class="text-custom-black dark:text-custom-white text-sm font-extrabold"
                  style="flex: 1; margin-left: 8px;"
                >
                  Account2
                </div>
                <div
                  class="text-custom-black dark:text-custom-white text-sm font-regular"
                  style="flex: 1; margin-left: 8px;"
                >
                  $
                  0
                </div>
              </button>
              <button
                aria-expanded="false"
                aria-haspopup="true"
                class="relative"
                data-testid="Account2-more-options"
                id="headlessui-listbox-button-:r2:"
                type="button"
              >
                <button
                  class="rounded-full p-2 hover:bg-custom-grey10 dark:hover:bg-custom-white10 sort-btn"
                  type="button"
                >
                  <more-horizontal.svg />
                </button>
              </button>
              <div
                style="position: absolute; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px; display: none;"
              />
            </div>
            <div
              class="MuiBox-root css-ytqq2v"
            >
              <div
                class="css-tac3yh"
              >
                <svg
                  class="inline-block select-none shrink-0 fill-custom-black dark:fill-custom-white text-4xl"
                  style="height: 1em; width: 1em;"
                >
                  <plus-circle.svg />
                </svg>
              </div>
              <div
                class="MuiBox-root css-hboir5"
              >
                <div
                  class="text-custom-black dark:text-custom-white text-sm font-extrabold"
                  style="flex: 1; margin-left: 8px;"
                >
                  Account.createSubAccount
                </div>
              </div>
            </div>
            <div
              class="MuiBox-root css-ytqq2v"
            >
              <div
                class="css-tac3yh"
              >
                <svg
                  class="inline-block select-none shrink-0 fill-custom-black dark:fill-custom-white text-4xl"
                  style="height: 1em; width: 1em;"
                >
                  <key.svg />
                </svg>
              </div>
              <div
                class="MuiBox-root css-hboir5"
              >
                <div
                  class="text-custom-black dark:text-custom-white text-sm font-extrabold"
                  style="flex: 1; margin-left: 8px;"
                >
                  Account.importAccount
                </div>
              </div>
            </div>
          </div>
          <div
            class="p-4 flex"
          >
            <button
              class="w-full border-[1px] border-solid border-primary dark:border-custom-white40 dark:hover:border-primary rounded-xl profile-btn flex justify-center items-center mr-1 dark:text-custom-white font-bold hover:bg-custom-grey10 dark:hover:bg-transparent p-3 rounded-md text-custom-black dark:text-custom-white"
              data-aid="logOutButton"
              type="button"
            >
              Actions.logout
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
